std/experimental/xml/appender.d

```diff
@@ -0,0 +1,198 @@
+/*
+*             Copyright Lodovico Giaretta 2016 - .
+*  Distributed under the Boost Software License, Version 1.0.
+*      (See accompanying file LICENSE_1_0.txt or copy at
+*            http://www.boost.org/LICENSE_1_0.txt)
+*/
+
+/*
+*   Appender implementation that uses a custom allocator. Meant for internal usage.
+*/
+
+module std.experimental.xml.appender;
+
+/*
+*   Appender implementation that uses a custom allocator. Meant for internal usage.
+*/
+package struct Appender(T, Alloc)
+{
+    import std.experimental.allocator;
+    import std.array;
+    import std.range.primitives;
+    import std.traits;
+
+    Alloc* allocator;
+    private Unqual!T[] arr;
+    private size_t used;
+
+    public this(ref Alloc alloc)
+    {
+        allocator = &alloc;
+    }
+    public this(Alloc* alloc)
```

Comment:
Can you define `@disable this();`? This is just in case someone would use the appender in the future and to prevent any misuse.